lib/locators.ts

```diff
@@ -0,0 +1,462 @@
+import * as util from 'util';
+let webdriver: any = require('selenium-webdriver');
+let clientSideScripts: any = require('./clientsidescripts');
+
+/**
+ * webdriver's By is an enum of locator functions, so we must set it to
+ * a prototype before inheriting from it.
+ */
+export class WebdriverBy {};
+WebdriverBy.prototype = webdriver.By;
+
+export interface ProtractorLocator extends webdriver.Locator {
```

Comment:
Since the webdriver Locator is namespaced, can we just `export interface Locator`? There should be no ambiguity since it would be imported from protractor.